sn_node/src/put_validation.rs

```diff
@@ -470,37 +467,23 @@ impl Node {
                 };
 
                 // check the spend
-                if let Err(e) = spends_with_parent
-                    .signed_spend
-                    .verify(spends_with_parent.signed_spend.spent_tx_hash())
-                {
+                if let Err(e) = signed_spend.verify(signed_spend.spent_tx_hash()) {
                     return Err(ProtocolError::InvalidSpendSignature(format!(
                         "while verifying spend for {:?}: {e:?}",
-                        spends_with_parent.signed_spend.dbc_id()
+                        signed_spend.dbc_id()
                     )));
                 }
 
-                // check parent tx hash
-                if spends_with_parent.parent_tx.hash()
-                    != spends_with_parent.signed_spend.dbc_creation_tx_hash()
-                {
-                    return Err(ProtocolError::InvalidSpendParents(format!(
-                    "parent tx hash does not match the parent tx in the DBC we're trying to spend: {:?} != {:?}",
-                    spends_with_parent.signed_spend.dbc_creation_tx_hash(),
-                    spends_with_parent.parent_tx.hash(),
-                )));
-                }
-
```

Comment:
This check can be safely removed, as `dbc_creation_tx_hash()` is now derived from the `signed_spend.spend.dbc_creation_tx.hash()`